steenroder/__init__.py

```diff
@@ -31,19 +31,21 @@ def get_coboundary(filtration):
     return coboundary
 
 
+@njit
 def get_reduced_triangular(matrix, homology=False):
     '''R = MV'''
 
-    # if a filtration is passed
-    if isinstance(matrix, tuple):
-        matrix = get_boundary(matrix)
-        if not homology:
-            matrix = np.flip(matrix, axis=[0, 1]).transpose()
+    # # if a filtration is passed
+    # if isinstance(matrix, tuple):
+    #     matrix = get_boundary(matrix)
+    #     if not homology:
+    #         matrix = np.flip(matrix, axis=[0, 1]).transpose()
 
     # reduction steps
     n = matrix.shape[1]
-    reduced = np.array(matrix)
-    triangular = np.eye(n, dtype=np.bool)
+    reduced = matrix.copy()
+    triangular = np.zeros((n, n), dtype=np.bool_)
```

Comment:
Inside `numba`, "real" numpy dtypes must be used instead of python in-built types such as `bool`, `float`, etc.  Note that `np.bool` is just an alias for `bool`, it is `np.bool_` which is the *real* numpy dtype.